l5kit/l5kit/data/map_api.py

```diff
@@ -0,0 +1,266 @@
+        return xyz
+
+    @staticmethod
+    @no_type_check
+    def is_lane(element: MapElement) -> bool:
+        """
+        Check whether an element is a valid lane
+
+        Args:
+            element (MapElement): a proto element
+
+        Returns:
+            bool: True if the element is a valid lane
+        """
+        return bool(element.element.HasField("lane"))
+
+    @lru_cache(maxsize=CACHE_SIZE)
+    def get_lane_coords(self, element_id: str) -> dict:
+        """
+        Get XYZ coordinates in world ref system for a lane given its id
+        lru_cached for O(1) access
+
+        Args:
+            element_id (str): lane element id
+
+        Returns:
+            dict: a dict with the two boundaries coordinates as (Nx3) XYZ arrays
+        """
+        element = self[element_id]
+        assert self.is_lane(element)
+
+        lane = element.element.lane
+        left_boundary = lane.left_boundary
+        right_boundary = lane.right_boundary
+
+        xyz_left = self.unpack_deltas_cm(
+            left_boundary.vertex_deltas_x_cm,
+            left_boundary.vertex_deltas_y_cm,
+            left_boundary.vertex_deltas_z_cm,
+            lane.geo_frame,
+        )
+        xyz_right = self.unpack_deltas_cm(
+            right_boundary.vertex_deltas_x_cm,
+            right_boundary.vertex_deltas_y_cm,
+            right_boundary.vertex_deltas_z_cm,
+            lane.geo_frame,
+        )
+
+        return {"xyz_left": xyz_left, "xyz_right": xyz_right}
+
+    @staticmethod
+    @no_type_check
+    def is_crosswalk(element: MapElement) -> bool:
+        """
+        Check whether an element is a valid crosswalk
+
+        Args:
+            element (MapElement): a proto element
+
+        Returns:
+            bool: True if the element is a valid crosswalk
+        """
+        if not element.element.HasField("traffic_control_element"):
+            return False
+        traffic_element = element.element.traffic_control_element
+        return bool(traffic_element.HasField("pedestrian_crosswalk") and traffic_element.points_x_deltas_cm)
+
+    @lru_cache(maxsize=CACHE_SIZE)
+    def get_crosswalk_coords(self, element_id: str) -> dict:
+        """
+        Get XYZ coordinates in world ref system for a crosswalk given its id
+        lru_cached for O(1) access
+
+        Args:
+            element_id (str): crosswalk element id
+
+        Returns:
+            dict: a dict with the polygon coordinates as an (Nx3) XYZ array
+        """
+        element = self[element_id]
+        assert self.is_crosswalk(element)
+        traffic_element = element.element.traffic_control_element
+
+        xyz = self.unpack_deltas_cm(
+            traffic_element.points_x_deltas_cm,
+            traffic_element.points_y_deltas_cm,
+            traffic_element.points_z_deltas_cm,
+            traffic_element.geo_frame,
+        )
+
+        return {"xyz": xyz}
+
+    def is_traffic_face_red(self, element_id: str) -> bool:
```

Comment:
Same logic as before in Oliver's comment. Why have three functions for the different faces if you can have one (e.g. determine_traffic_light_face or similar)? It seems that this boils down to manipulating the string '_green_face' to just a variable color +'_face' ?

Reply:
mainly to avoid calling it with `green` or `Green` or `GREEN` or `GReen` :) Once we switch to enum I'm fine in replacing them with just a single function

Reply:
I'll go on and do it since there is quorum on this. We will refactor it later to support enum

Reply:
So you're sticking to the current implementation or go with the single method? Sorry, just couldn't figure out which one you meant from the reply :)

Reply:
single one :)